aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -1027,11 +1080,55 @@ module aptos_framework::delegation_pool {
         });
     }
 
+    /// Allows an owner to update the commission percentage for the operator of the underlying stake pool.
+    public entry fun update_commission_percentage(
+        owner: &signer,
+        new_commission_percentage: u64
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords, BeneficiaryForOperator, NextCommissionPercentage {
+        assert!(features::commission_change_delegation_pool_enabled(), error::invalid_state(
+            ECOMMISSION_RATE_CHANGE_NOT_SUPPORTED
+        ));
+        assert!(new_commission_percentage <= MAX_FEE, error::invalid_argument(EINVALID_COMMISSION_PERCENTAGE));
+        let pool_address = get_owned_pool_address(signer::address_of(owner));
+        assert!(
+            operator_commission_percentage(pool_address) + MAX_COMMISSION_INCREASE >= new_commission_percentage,
+            error::invalid_argument(ETOO_LARGE_COMMISSION_INCREASE)
+        );
+        assert!(
+            stake::get_remaining_lockup_secs(pool_address) >= min_remaining_secs_for_commission_change(),
+            error::invalid_state(ETOO_LATE_COMMISSION_CHANGE)
+        );
+
+        // synchronize delegation and stake pools before any user operation
+        // ensure the operator is paid its uncommitted commission rewards
+        // with the old commission percentage
+        synchronize_delegation_pool(pool_address);
+
+        let delegation_pool = borrow_global<DelegationPool>(pool_address);
```

Comment:
would move this resource read to the else branch

Reply:
Done!